ui-particles-angular/projects/ui-policy-studio-angular/src/lib/gio-policy-studio.component.ts

```diff
@@ -0,0 +1,212 @@
+import { PolicyServiceAbstract } from './services/policy.abstract.service';
+import { ResourceServiceAbstract } from './services/resource.abstract.service';
+import { SpelServiceAbstract } from './services/spel.abstract.service';
+import { Grammar } from './models/spel';
+
+interface UrlParams {
+  path: string;
+  tabId: string;
+  flowsIds: string[];
+}
+
+@Component({
+  selector: 'gio-policy-studio',
+  templateUrl: './gio-policy-studio.component.html',
+  styleUrls: ['./gio-policy-studio.component.scss'],
+})
+export class GioPolicyStudioComponent implements OnInit {
+  @Input()
+  // TODO: Remove ! and add init value ! for all
+  canAdd!: boolean;
+
+  @Input()
+  canDebug!: boolean;
+
+  @Input()
+  hasResources!: boolean;
+
+  @Input()
+  hasProperties!: boolean;
+
+  @Input()
+  hasPolicyFilter!: boolean;
+
+  @Input()
+  sortable!: boolean;
+
+  @Input()
+  set policies(policies: PolicyListItem[]) {
+    this._policies = chain(policies)
+      .map(policy => (!isNil(policy.category) ? policy : { ...policy, category: this.unknownPolicyCategory }))
+      // First sort by category (based on category order) and then by name
+      .sortBy([policy => this.policyCategoriesOrder.indexOf(policy.category ?? ''), policy => policy.name])
+      .value();
+  }
+
+  get policies() {
+    return this._policies;
+  }
+
+  @Input()
+  definition: unknown;
+
+  @Input()
+  services: Record<string, unknown> = {};
+
+  @Input()
+  flowSchema: unknown;
+
+  @Input()
+  resourceTypes!: unknown[];
+
+  @Input()
+  propertyProviders!: unknown[];
+
+  @Input()
+  readonlyPlans!: boolean;
+
+  @Input()
+  dynamicPropertySchema: unknown = {};
+
+  @Input()
+  debugResponse: unknown;
+
+  @Input()
+  flowsTitle!: string;
+
+  @Input()
+  hasConditionalSteps!: boolean;
+
+  @Input()
+  configurationInformation =
+    'By default, the selection of a flow is based on the operator defined in the flow itself. This operator allows either to select a flow when the path matches exactly, or when the start of the path matches. The "Best match" option allows you to select the flow from the path that is closest.';
+
+  @Output()
+  save = new EventEmitter<unknown>();
+
+  tabId!: string;
+  selectedFlowsIds!: string;
+  configurationSchema!: FlowConfigurationSchema;
+  policyDocumentation!: { id: string; image: string; content: string };
+
+  private readonly unknownPolicyCategory = 'others';
+  private readonly policyCategoriesOrder = ['security', 'performance', 'transformation', this.unknownPolicyCategory];
+  private _policies!: PolicyListItem[];
+
+  constructor(
+    private readonly location: Location,
+    @Inject('FlowService') private readonly flowService: FlowServiceAbstract,
+    @Inject('PolicyService') private readonly policyService: PolicyServiceAbstract,
+    @Inject('ResourceService') private readonly resourceService: ResourceServiceAbstract,
+    @Inject('SpelService') private readonly spelService: SpelServiceAbstract,
+  ) {}
+
+  ngOnInit(): void {
```

Comment:
public ? reading other method with visibility and only this without any is strange to me ^^

Reply:
let me find the eslint rule and ok 🚀